packages/json-rpc-engine/src/JsonRpcEngine.ts

```diff
@@ -211,15 +217,26 @@ export class JsonRpcEngine extends SafeEventEmitter {
 
     if (Array.isArray(req)) {
       if (callback) {
-        return this.#handleBatch(req, callback);
+        return this.#handleBatch(
+          req,
+          // This assertion is safe because of the runtime checks validating that `req` is an array and `callback` is defined.
+          // There is only one overload signature that satisfies both conditions, and its `callback` type is the one that's being asserted.
+          callback as (
+            error: unknown,
+            responses?: JsonRpcResponse<Json>[],
+          ) => void,
```

Comment:
Ideally, TypeScript would have better pattern-matching around overload signatures and would be capable of narrowing `callback` and `responses` so that this assertion is unnecessary.